solidity/contracts/libs/OwnableMOfNSet.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ External Imports ============
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+// ============ Internal Imports ============
+import {EnumerableMOfNSet} from "./EnumerableMOfNSet.sol";
+import {Message} from "./Message.sol";
+
+contract OwnableMOfNSet is Ownable {
+    // ============ Libraries ============
+
+    using Message for bytes;
+
+    // ============ Mutable Storage ============
+
+    /// @notice The set for each remote domain.
+    mapping(uint32 => EnumerableMOfNSet.AddressSet) private _sets;
```

Comment:
imo this contract should be specific to a domain and have the ISM to the routing

Reply:
Oh interesting. So rather than *inherit* from OwnableMOfNset in AggregationISM, you would have AggregationISM point to a bunch of deployed OwnableMOfNSet?

Reply:
I think so?
intuitively I wouldn't expect an `OwnableMofNSet` to have any domain key stuff, that seems higher level

Reply:
So ignoring the name for a moment, do you think the right approach is for AggregationIsm to basically route to a different contract for each domain?